src/pages/auth/SignIn.jsx

```diff
@@ -1,7 +1,101 @@
-import React from 'react';
+import { Link, useNavigate } from 'react-router-dom';
+import React, { useState } from 'react';
+
+import { login } from '../../api/DBRequests';
 
 const SignIn = () => {
-  return <></>;
+  const [form, setForm] = useState({ email: '', password: '' });
+  const [error, setError] = useState({});
+  const navigate = useNavigate();
+
+  const handleChange = (e) => {
+    setForm({ ...form, [e.target.name]: e.target.value });
+  };
+
+  const handleSubmit = async (e) => {
```

Comment:
You could extract validating errors to a separate function:
```js
const validateForm = () => {
    const errors = {};
    if (!form.email) errors.email = 'Email is required';
    if (!form.password) errors.password = 'Password is required';
    return errors;
  };
```
In that case, you would have to change your `handleSubmit` function:
```js
const handleSubmit = async (e) => {
    e.preventDefault();

    const validationErrors = validateForm();
    if (Object.keys(validationErrors).length > 0) {
      setError(validationErrors);
      return;
    }

    try {
      const result = await login(form);
      if (result.status === 200) {
        navigate('/');
      }
    } catch {
      setError((prevError) => ({
        ...prevError,
        form: 'Invalid email or password',
      }));
    }
  };
```

Or you could do it inside of `handleSubmit` function:
```js
const handleSubmit = (e) => {
  e.preventDefault();

  const errors = {};
  if (!form.email) errors.email = 'Email is required';
  if (!form.password) errors.password = 'Password is required';

  if (Object.keys(errors).length > 0) {
    setError(errors);
    return;
  }

  ...
};
```
Although, none of that is necessary and you could keep it as it is.

Reply:
```
const validationErrors = validateForm();
setError(validationErrors);
if (Object.keys(validationErrors).length > 0) {   
  return;
}
```

I did it this way, because setError also resets the errors.